cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,342 @@
+}
+
+/**
+ * Get specific field of DocTable.
+ * @param {number} columnNumber Integer starts from 0 for the first column
+ * @param {number} rowNumber Integer starts from 0 for the first row
+ */
+export function getDocTableField(columnNumber, rowNumber) {
+  return getDocTableRow(rowNumber).findElementByTestId('docTableField').eq(columnNumber);
+}
+
+/**
+ * find all Rows in Doc Table Field Expanded Document.
+ * @param expandedDocument cypress representation of the Doc Table Field Expanded Document
+ */
+export function findDocTableExpandedDocRowsIn(expandedDocument) {
+  return expandedDocument.findElementByTestIdLike('tableDocViewRow-');
+}
+
+/**
+ * Get the "expandedDocumentRowNumber"th row from the expanded document from the "docTableRowNumber"th row of the DocTable.
+ * @param {number} docTableRowNumber Integer starts from 0 for the first row
+ * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+ * @example
+ * // returns the first row from the expanded document from the second row of the DocTable.
+ * getExpandedDocRow(1, 0);
+ */
+export function getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber) {
+  return findDocTableExpandedDocRowsIn(getDocTableRow(docTableRowNumber + 1)).eq(
+    expandedDocumentRowNumber
+  );
+}
+
+/**
+ * Get the value for the "expandedDocumentRowNumber"th row from the expanded document from the "docTableRowNumber"th row of the DocTable.
+ * @param {number} docTableRowNumber Integer starts from 0 for the first row
+ * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+ * @example
+ * // returns the value of the field from the first row from the expanded document from the second row of the DocTable.
+ * getExpandedDocRow(1, 0);
+ */
+export function getExpandedDocRowValue(docTableRowNumber, expandedDocumentRowNumber) {
+  return getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber)
+    .find(`[data-test-subj*="tableDocViewRow-"]`)
+    .find('span');
+}
+
+/**
+ * Get the field name for the "expandedDocumentRowNumber"th row from the expanded document from the "docTableRowNumber"th row of the DocTable.
+ * @param {number} docTableRowNumber Integer starts from 0 for the first row
+ * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+ * @example
+ * // returns the name of the field from the first row from the expanded document from the second row of the DocTable.
+ * getExpandedDocRow(1, 0);
+ */
+export function getExpandedDocRowFieldName(docTableRowNumber, expandedDocumentRowNumber) {
+  return getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber)
+    .find('td')
+    .eq(1) // Field name is in the second column.
+    .find('span[class*="textTruncate"]');
+}
+
+/**
+ * Select a language in the Dataset Selector for Index
+ * @param {string} datasetLanguage Index supports "OpenSearch SQL" and "PPL"
+ */
+export function selectIndexDatasetLanguage(datasetLanguage) {
```

Comment:
fyi lot of these i will include as global commands in my PR (ones related to setting dataset)